config/routes/api/v2.rb

```diff
@@ -221,7 +225,10 @@ class ActionDispatch::Routing::Mapper
           end
         end
         api_resources :puppet_modules, :only => [:index, :show] do
-          get :auto_complete_search, :on => :collection
+          collection do
+            get :compare
```

Comment:
I don't entirely follow. Looks like you already have code in place to handle that difference?